components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -989,6 +994,15 @@ public boolean testRead(String[] args)
     throws FormatException, ServiceException, IOException
   {
     DebugTools.enableLogging("INFO");
+
+    if (args.length == 1 && HELP_ARGUMENTS.contains(args[0])) {
```

Comment:
It can be done in a follow-up if needed, but it probably makes sense to do what @joshmoore suggests, which means that ```parseArgs``` is never called if a help argument is supplied (so it shouldn't be a complete rewrite, just changing two lines).

The current behavior is fine if you follow the rules and only have one argument that is the help argument, but something simple like ```showinf -h /path/to/file``` is going to terminate early (due to https://github.com/mtbc/bioformats/blob/trac-12954-usage-help-argument/components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java#L273) without performing either of the requested actions.  Always printing the help and exiting if the user is mixing help/non-help arguments seems the safest bet.